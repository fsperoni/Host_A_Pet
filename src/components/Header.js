import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import UserContext from "./UserContext";
import logoImg from '../assets/logo.png';
import '../styles/Header.scss';

const Header = ({ logout }) => {
  const { currentUser } = useContext(UserContext);

  const signIn = () => {
    return(
      <>
      <NavLink end to="/login">
        Login
      </NavLink>
      <NavLink className="ms-3" end to="/signup">
        Sign Up
      </NavLink>
      </>
    )
  }

  const signOut = () => {
    return (
      <>
      <span className="me-3">Hello {currentUser.first_name || currentUser.username}!</span>
      <Link to="/" onClick={logout}>
        Log out
      </Link>
      </>
    )
  }

  return (
    <Navbar expand="sm" direction="horizontal">
      <NavLink end to="/" className="navbar-brand">
        <div className="d-flex">
          <img src={logoImg} alt="Host a Pet"/>
          <h1 className="my-auto">Host a Pet</h1>
        </div>
      </NavLink>
      <Nav navbar className="ms-5">
        <NavItem>
          <NavLink className="mr-2" end to="/">Home</NavLink>
        </NavItem>
        <NavItem>
          <NavLink className="mr-2" end to="/newBooking">New Booking</NavLink>
        </NavItem>
        <NavItem>
          <NavLink className="mr-2" end to="/myBookings">My Bookings</NavLink>
        </NavItem>
        <NavItem>
          <NavLink className="mr-2" end to="/myPets">My Pets</NavLink>
        </NavItem>
        <NavItem>
          <NavLink end to="/myProfile">My Profile</NavLink>
        </NavItem>
      </Nav>
      <div className="login">
        {currentUser ? signOut() : signIn()}
      </div> 
    </Navbar>
  );
}

export default Header;