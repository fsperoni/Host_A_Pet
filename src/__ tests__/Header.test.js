import { render } from "@testing-library/react";
import Header from "../components/Header";
import { BrowserRouter } from "react-router-dom";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser,
} = require("./_testCommon");

// smoke test
it('renders without crashing', function () {
  render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: currentAdminUser(), function() { } }}>
        <Header logout={function () { }} />
      </UserContext.Provider>
    </BrowserRouter>);
});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: currentAdminUser(), function() { } }}>
        <Header logout={function () { }} />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Host a Pet' in the component.", function () {
  const { getByText } = render(
    <BrowserRouter>
      <UserContext.Provider
        value={{ currentUser: currentAdminUser(), function() { } }}>
        <Header logout={function () { }} />
      </UserContext.Provider>
    </BrowserRouter>
  );
  expect(getByText("Host a Pet")).toBeInTheDocument();
});
