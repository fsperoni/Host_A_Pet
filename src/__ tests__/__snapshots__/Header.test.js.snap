// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-sm"
    direction="horizontal"
  >
    <div
      class="container-fluid"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        <div
          class="d-flex"
        >
          <img
            alt="Host a Pet"
            src="logo.png"
          />
          <h1
            class="my-auto"
          >
            Host a Pet
          </h1>
        </div>
      </a>
      <ul
        class="ms-5 navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="mr-2 active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="mr-2"
            href="/newBooking"
          >
            New Booking
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="mr-2"
            href="/myBookings"
          >
            My Bookings
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="mr-2"
            href="/myAvailabilities"
          >
            My Availabilities
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="mr-2"
            href="/myPets"
          >
            My Pets
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class=""
            href="/myProfile"
          >
            My Profile
          </a>
        </li>
      </ul>
      <div
        class="login"
      >
        <span
          class="me-3"
        >
          Hello testUser!
        </span>
        <a
          href="/"
        >
          Log out
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
